WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,136 @@ import {
   List,
 } from 'semantic-ui-react';
 import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
 import {
   roleListUrl,
+  currentUserUrl,
+  teamUrl,
 } from '../../lib/requests/routes.js.erb';
 import Errored from '../Requests/Errored';
 import Loading from '../Requests/Loading';
 import RoleForm from './RoleForm';
+import I18n from '../../lib/i18n';
+
+// let i18n-tasks know the key is used
+// i18n-tasks-use t('enums.user.role_status.trainee_delegate')
+// i18n-tasks-use t('enums.user.role_status.candidate_delegate')
+// i18n-tasks-use t('enums.user.role_status.delegate')
+// i18n-tasks-use t('enums.user.role_status.senior_delegate')
+// i18n-tasks-use t('enums.user.role_status.member')
+// i18n-tasks-use t('enums.user.role_status.senior_member')
+// i18n-tasks-use t('enums.user.role_status.leader')
+
+export default function RolesTab({ userId }) {
+  const loggedInUserFetch = useLoadedData(currentUserUrl);
+  const roleListFetch = useLoadedData(roleListUrl(userId));
+  const { data, sync } = roleListFetch;
+  const { save, saving } = useSaveAction();
 
-function RoleFormModal({
-  trigger, userId, isActiveRole, sync,
-}) {
   const [open, setOpen] = React.useState(false);
+  const [currentUserRoles, setCurrentUserRoles] = React.useState(null);
+  const [error, setError] = React.useState(null);
 
-  return (
-    <Modal
-      size="fullscreen"
-      onClose={() => {
-        setOpen(false);
-        sync();
-      }}
-      onOpen={() => setOpen(true)}
-      open={open}
-      trigger={trigger}
-    >
-      <Modal.Content>
-        <RoleForm
-          userId={userId}
-          isActiveRole={isActiveRole}
-        />
-      </Modal.Content>
-    </Modal>
+  const activeNonHiddenRoles = React.useMemo(
+    () => {
+      if (data) {
+        return data.activeRoles.filter((role) => role.group.is_hidden === false);
+      }
+      return [];
+    },
+    [data],
+  );
+  const isDelegate = React.useMemo(
+    () => {
+      if (data) {
+        return data.activeRoles.some((role) => role.group.group_type === 'delegate_regions');
+      }
+      return false;
+    },
+    [data],
   );
-}
 
-export default function RolesTab({ userId }) {
-  const {
-    data, sync, loading, error,
-  } = useLoadedData(roleListUrl(userId));
+  React.useEffect(() => {
+    if (!loggedInUserFetch.loading) {
```

Comment:
We shouldn't need this if check on loading. Remove this line and the loggedInUserFetch.loading and the code should work all the same.

`useEffect` only runs when the dependencies update. If those dependencies are just "save" and loggedInUserFetch.data, then this will just update when those change.

Reply:
Okay thanks Cailyn. But after some optimizations, this useEffect has been removed.